iModelCore/ECDb/Tests/NonPublished/ECSqlStatementTests.cpp

```diff
@@ -11036,42 +11036,97 @@ TEST_F(ECSqlStatementTestFixture, verify_inf_and_nan_handling) {
+        stmt.BindPoint3d(8,  DPoint3d::From(neg_inf, neg_inf, neg_inf)); // ECSQL treat negative inf as null
+        stmt.BindPoint3d(9,  DPoint3d::From(nan_val, nan_val, nan_val)); // ECSQL treat nan as null
         ASSERT_EQ(stmt.Step(), BE_SQLITE_DONE);
     }
     if("read row") {
         ECSqlStatement stmt;
-        ASSERT_EQ(stmt.Prepare(m_ecdb, "select inf_pos, inf_neg, nan_val from ts.Element"), ECSqlStatus::Success);
+        ASSERT_EQ(stmt.Prepare(m_ecdb, selectStmt), ECSqlStatus::Success);
         ASSERT_EQ(stmt.Step(), BE_SQLITE_ROW);
-        auto inf_pos = stmt.GetValueDouble(0);
-        auto inf_neg = stmt.GetValueDouble(1);
-        auto nan_val = stmt.GetValueDouble(2);
 
-        ASSERT_EQ(std::isinf(inf_pos), false);
-        ASSERT_EQ(std::isinf(inf_neg), false);
-        ASSERT_EQ(std::isnan(nan_val), false);
+        for (auto columnIndex = 0; columnIndex < stmt.GetColumnCount(); ++columnIndex)
+            {
+            switch (columnIndex / 3)
+                {
+                case 1:
+                    {
+                    // Test Point2D properties
+                    double coordX, coordY;
+                    stmt.GetValuePoint2d(columnIndex).GetComponents(coordX, coordY);
+
+                    // Check the +inf and -inf column values
+                    EXPECT_EQ(std::isinf(coordX), false) << "Value of co-ordinate X column " << columnIndex + 1 << " should not be infinity";
+                    EXPECT_EQ(std::isinf(coordY), false) << "Value of co-ordinate Y column " << columnIndex + 1 << " should not be infinity";
 
-        auto inf_pos_isnull = stmt.IsValueNull(0);
-        auto inf_neg_isnull = stmt.IsValueNull(1);
-        auto nan_val_isnull = stmt.IsValueNull(2);
+                    // Check the nan column values
+                    EXPECT_EQ(std::isnan(coordX), false) << "Value of co-ordinate X column " << columnIndex + 1 << " should not be NaN";
+                    EXPECT_EQ(std::isnan(coordY), false) << "Value of co-ordinate Y column " << columnIndex + 1 << " should not be NaN";
 
-        ASSERT_EQ(inf_pos_isnull, true);
-        ASSERT_EQ(inf_neg_isnull, true);
-        ASSERT_EQ(nan_val_isnull, true);
+                    EXPECT_EQ(stmt.IsValueNull(columnIndex), true) << "Point2D at column " << columnIndex + 1 << " should be NULL";
+                    break;
+                    }
+                case 2:
+                    {
+                    // Test Point3D properties
+                    double coordX, coordY, coordZ;
+                    stmt.GetValuePoint3d(columnIndex).GetComponents(coordX, coordY, coordZ);
+                    // Check the +inf and -inf column values
+                    EXPECT_EQ(std::isinf(coordX), false) << "Value of co-ordinate X column " << columnIndex + 1 << " should not be infinity";
+                    EXPECT_EQ(std::isinf(coordY), false) << "Value of co-ordinate Y column " << columnIndex + 1 << " should not be infinity";

```

Comment:
You test do not include `Point2d(Inf,1)` & `Point2d(1, Inf)` and then check `IsValueNull()`. Generally Point2d is primitive value `X` and `Y` is either both `NULL` or both not `NULL`. But if you have one component is NULL and not other we return `false` from `IsValueNull()`.

I think the correct behavior is to insert/update to NULL for all value if any value is NULL/INF/NAN. And similarly return IsValueNull() true if any component is Null/Inf/Nan. But we do not have to do it right now.

Reply:
Okay, I'll add a test and check it.